### ISSUE TITLE
SearchForm2.php does not handle "enum" properly

### ISSUE THREAD

**Issue by DevlshOne** [salesagility/SuiteCRM#2852]
#### Issue
When performing a Search using the `subquery` operator in a module, the SearchForm2.php puts extra escaped quotes around an *enum* field.

#### Expected Behavior
Script should recognize an *enum* field type and remove the extra quotes.

#### Possible Fix
Allow for another parameter that tells this script whether or not to add quotes to the `field`.

#### Steps to Reproduce
1. Add something like this to **searchdefs.php** for your module:
```
            'iso_type_c' =>
            array(
                'label' => 'LBL_CONTRACT_ISO_TYPE',
                'type' => 'enum',
                'default' => true,
                'link' => false,
                'studio' => 'visible',
                'width' => '10%',
                'name' => 'iso_type_c',
            ),
```
2. Add soemthing like this to the **SearchFields.php** for your module:
```
    'iso_type_c' =>
    array(
        'query_type' => 'default',
        'operator' => 'subquery',
        'subquery' => 'SELECT id FROM nw_iso i WHERE i.deleted = 0 AND i.iso_type LIKE',
        'db_field' => array('nw_iso_id_c')
    ),
```
3. The resulting **raw SQL** looks like this:
`SELECT  nw_contract.id , nw_contract.nw_merchant_id_c , nw_contract.nw_iso_id_c , nw_contract.name , jt0.name merchant , jt1.name iso , nw_contract.purchase_amount , nw_contract.outstanding_balance , nw_contract.contract_status , nw_contract.underwriting_status , nw_contract.last_payment_date , nw_contract.funding_status , nw_contract.iso_type_c , nw_contract.assigned_user_id  FROM nw_contract  LEFT JOIN nw_contract_cstm ON nw_contract.id = nw_contract_cstm.id_c  LEFT JOIN nw_merchant jt0 ON nw_contract.nw_merchant_id_c = jt0.id AND jt0.deleted=0  LEFT JOIN nw_iso jt1 ON nw_contract.nw_iso_id_c = jt1.id AND jt1.deleted=0  where ((nw_contract.nw_iso_id_c IN (select  id  from (SELECT id FROM nw_iso i WHERE i.deleted = 0 AND i.iso_type LIKE '\'internal\'%') jt1.name_type_c_derived))) AND nw_contract.deleted=0 ORDER BY nw_contract.date_entered DESC LIMIT 0,31`

[NWE925.zip](https://github.com/salesagility/SuiteCRM/files/680770/NWE925.zip)
#### Your Environment
* SuiteCRM Version used: 7.4.3
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7): MariaDB, PHP 5.4
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 16.02

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [SearchForm2.php does not handle "enum" properly](https://trello.com/c/OtgcQ1Tg/43-searchform2-php-does-not-handle-enum-properly)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
